src/api/app/views/webui/users/notifications/_notifications_list.html.haml

```diff
@@ -21,46 +21,8 @@
-            .list-group-item.px-0.px-md-1.py-3
-              .row
-                .col-auto.pr-0
-                  .custom-control.custom-checkbox
-                    = check_box_tag('notification_ids[]', notification.id, false,
-                                    id: "notification_ids_#{notification.id}", class: 'custom-control-input')
-                    = label_tag("notification_ids_#{notification.id}", '', class: 'custom-control-label')
-                .col-10
-                  .row
-                    .col
-                      - if notification.notifiable_type == 'BsRequest'
-                        = image_tag('icons/request-icon.svg', height: 18, title: 'Request notification')
-                        = link_to(notification.notifiable_link[:text], notification.notifiable_link[:path], class: 'mx-1')
-                        %small.text-nowrap #{time_ago_in_words(notification.created_at)} ago
-                        %span.badge.ml-1{ class: "badge badge-#{request_badge_color(notification.notifiable.state)}" }
-                          = notification.notifiable.state
-                      - else
-                        %i.fas.fa-comments{ title: 'Comment notification' }
-                        = link_to(notification.notifiable_link[:text], notification.notifiable_link[:path], class: 'mx-1')
-                        %small.text-nowrap #{time_ago_in_words(notification.created_at)} ago
-                    .col-auto.actions.ml-auto.align-self-end.align-self-md-start
-                      :ruby
-                        title, icon = notification.unread? ? ['Mark as "Read"', 'fa-check'] : ['Mark as "Unread"', 'fa-undo']
-                        update_path = my_notifications_path(notification_ids: [notification.id], type: selected_filter[:type],
-                                                            project: selected_filter[:project], page: params[:page],
-                                                            show_more: params[:show_more])
-                      = link_to(update_path, id: format('update-notification-%d', notification.id),
-                                method: :put, class: 'btn btn-sm btn-outline-success px-3', title: title,
-                                data: { remote: true, 'disable-with': tag.i(class: 'fas fa-spinner fa-spin') }) do
-                        %i.fas{ class: "#{icon}" }
-                  .row.mt-1.pl-sm-4
-                    .col-auto.pr-0
-                      = render partial: 'notification_avatars', locals: { avatar_objects: notification.avatar_objects }
-                    .col-auto.pl-xs-2
-                      .smart-overflow= action_description(notification)
-                  .row.d-none.d-md-block.pl-4
-                    .col
-                      %p.mt-3.mb-0= notification.excerpt
+          - notifications.each do |notification|
+            = render NotificationComponent.new(notification, selected_filter)
```

Comment:
I think you can use the collections helpers in view controllers[^1] to render collections.

[^1]: https://viewcomponent.org/guide/collections.html

Reply:
True! I've changed the call to the component to use the collection instead.